src/nvme/ioctl.c

```diff
@@ -1237,7 +1237,20 @@ int nvme_ns_mgmt(struct nvme_ns_mgmt_args *args)
 {
 	__u32 cdw10    = NVME_SET(args->sel, NAMESPACE_MGMT_CDW10_SEL);
 	__u32 cdw11    = NVME_SET(args->csi, NAMESPACE_MGMT_CDW11_CSI);
-	__u32 data_len = args->ns ? sizeof(*args->ns) : 0;
+	__u32 data_len = 0;
+
+	if (args->ns) {
+		data_len = sizeof(*args->ns);
+		memcpy((__u8 *)args->ns + NVME_NS_MGMT_NPHNDLS,
+			&args->nphndls, sizeof(args->nphndls));
+		if (args->io_cmd_set_specific)
```

Comment:
so do you recommend using just number instead of using enum?
like below?
		memcpy((__u8 *)args->ns + 392,
			&args->nphndls, sizeof(args->nphndls));

Reply:
The idea is to use `sizeof` to figure out if the user passes in v1 or v2 of the data structure. If it is v2 than you just use the correct data types and avoid using `memcpy` with offsets.

Reply:
To not doing that, I modified commit as you recommended
Check it out